core/ir/instr_shared.c

```diff
@@ -3101,6 +3102,16 @@ instr_convert_to_isa_regdeps(void *drcontext, instr_t *instr_real_isa,
         for (uint reg = 0; reg < REGDEPS_MAX_NUM_REGS; ++reg) {
             if (dst_reg_used[reg]) {
                 opnd_t dst_opnd = opnd_create_reg((reg_id_t)reg);
+                /* Virtual registers may have ID values not supported by reg_get_size().
+                 * We set the opnd_t.size field to be the operation_size max_opnd_size so
+                 * that reg_get_size() can return it without triggering a CLIENT_ASSERT
```

Comment:
Reading these 1st 2 sentences I am thinking "just go fix reg_get_size: why try to work around it here?  why are you treating reg_get_size like some read-only unchangeable thing?".  The key is the 3rd sentence that the sizes are not fixed: put that up front in the 1st sentence to make it clear we can't easily have reg_get_size support virtual registers.

Reply:
Right, done.